src/config.ts

```diff
@@ -0,0 +1,86 @@
+import {AuthenticationConfig, ServiceConfig} from '../third_party/types/common-types';
+
+// ProfileAgentConfig is the configuration for the profiler.
+export interface ProfileAgentConfig extends AuthenticationConfig {
+  // Log levels: 0-disabled,1-error,2-warn,3-info,4-debug.
+  logLevel?: number;
+
+  // Specifies the service contect with which profiles from this application
+  // will be associated.
+  serviceContext?: {
+    // The service name.
+    service?: string;
+
+    // The service version.
+    version?: string;
+
+    // Unique deployment identifier.
+    minorVersion?: string;
+  };
+
+  // Configuration for heap profiling.
+  heapProfilerConfig?: HeapProfilerConfig;
+
+  // Configuration for cpu profiling.
+  cpuProfilerConfig?: CpuProfilerConfig;
+
+  instance?: string;
+  zone?: string;
+}
+
+// HeapProfilerConfig is the configuration for the heap profiler.
+export interface HeapProfilerConfig {
+  // When disable is true, heap profiling will be disabled.
+  disable?: boolean;
+
+  // The maximum number of stack frames to be captured on each allocation.
+  stackDepth?: number;
+
+  // Sampling interval for Heap profiler, specified in number of bytes.
+  samplingInterval?: number;
+}
+
+// CpuProfilerConfig is the configuration for the cpu profiler.
+export interface CpuProfilerConfig {
+  // When disable is true, cpu profiling will be disabled.
+  disable?: boolean;
+
+  // Sampling interval for CPU profiler, specified in number of microseconds.
+  samplingInterval?: number;
+}
+
+export const defaultHeapProfilerConfig: HeapProfilerConfig = {
+  disable: false,
+  stackDepth: 32,
+  samplingInterval: 1024 * 1024
+};
+
+export const defaultCpuProfilerConfig: CpuProfilerConfig = {
+  disable: false,
+  samplingInterval: 1000,
+};
+
+export const defaultAgentConfig: ProfileAgentConfig = {
```

Comment:
It's a bit strange that this default config is in the interface file. I would expect passing something like an empty struct makes a default config. How would it look like to pass a config where I want to set just one non-default attribute?

We generally need some examples of using the API here. Perhaps get an README.md file.

Reply:
Normally, the JS api accepting the config would 'extend' the default config object with the user provided config object. Here's an [example](https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/blob/7af102e5218b94c79545c40391fa88024fcc0c0e/index.ts#L78) of how this works in the trace agent. This is a common idiom.

From the API user's point of view, they provide only the values they need to customize. Everything else is inherited.